govcd/admincatalog.go

```diff
@@ -84,3 +84,39 @@ func (adminCatalog *AdminCatalog) Refresh() error {
 
 	return nil
 }
+
+// getOrgInfo finds the organization to which the admin catalog belongs, and returns its name and ID
+func (adminCatalog *AdminCatalog) getOrgInfo() (orgInfoType, error) {
+	previous, exists := orgInfoCache[adminCatalog.AdminCatalog.ID]
+	if exists {
+		return previous, nil
+	}
+	var orgId string
+	var orgHref string
+	var err error
+	for _, link := range adminCatalog.AdminCatalog.Link {
+		if link.Rel == "up" && (link.Type == types.MimeOrg || link.Type == types.MimeAdminOrg) {
+			orgId, err = GetUuidFromHref(link.HREF, true)
+			if err != nil {
+				return orgInfoType{}, err
+			}
+			orgHref = link.HREF
+			break
+		}
+	}
+	if orgHref == "" || orgId == "" {
+		return orgInfoType{}, fmt.Errorf("error retrieving org info for admin catalog %s", adminCatalog.AdminCatalog.Name)
+	}
+	var org types.Org
+	_, err = adminCatalog.client.ExecuteRequest(orgHref, http.MethodGet,
+		"", "error retrieving org: %s", nil, &org)
+	if err != nil {
+		return orgInfoType{}, err
+	}
+
+	orgInfoCache[adminCatalog.AdminCatalog.ID] = orgInfoType{
+		id:   orgId,
```

Comment:
We can avoid orgId parsing from href and just use `org.Id`

Reply:
We need the bare UUID.
It will be used to create the context related header, which requires the format 
```
X-VMWARE-VCLOUD-TENANT-CONTEXT: [19f2dc0a-fd92-46c4-a2ca-bb0a95411093]
```